test/L2/L2AirdropV2.t.sol

```diff
@@ -0,0 +1,594 @@
+
+    function test_ClaimAirdrop_FullAirdrop() public {
+        // alice satisfies staking tier 1 condition
+        aliceSatifiesStakingTier1();
+
+        // alice satisfies staking tier 2 condition
+        aliceSatifiesStakingTier2();
+
+        uint256 aliceBalanceBefore = l2LiskToken.balanceOf(alice);
+        bytes32[] memory merkleProof = new bytes32[](1);
+        merkleProof[0] = bytes32(0xf0df3dcda05b4fbd9c655cde3d5ceb211e019e72ec816e127a59e7195f2cd7f5);
+
+        // check that the AirdropClaimed event is emitted for all conditions
+        vm.expectEmit(true, true, true, true);
+        emit L2AirdropV2.AirdropClaimed(
+            aliceLSKAddress, 80 * 10 ** 18, alice, l2AirdropV2.STAKING_TIER_1_BIT() | l2AirdropV2.STAKING_TIER_2_BIT()
+        );
+
+        l2AirdropV2.claimAirdrop(aliceLSKAddress, 80 * 10 ** 18, merkleProof);
+        assertEq(l2LiskToken.balanceOf(alice), aliceBalanceBefore + 80 * 10 ** 18); // 16 L2LiskToken airdrop
+
+        // check airdrop claim status for alice
+        assertEq(l2AirdropV2.claimedStakingTier1(bytes20(alice)), true);
+        assertEq(l2AirdropV2.claimedStakingTier2(bytes20(alice)), true);
+        assertEq(l2AirdropV2.claimedFullAirdrop(bytes20(alice)), true);
+
+        // check that alice cannot claim airdrop again
+        vm.expectRevert("L2AirdropV2: full airdrop claimed");
+        l2AirdropV2.claimAirdrop(bytes20(alice), 80 * 10 ** 18, merkleProof);
+    }
+
+    function test_ClaimAirdrop_NotStartedYet() public {
+        // re-deploy L2AirdropV2 contract because merkle root is already set in setup
+        l2AirdropV2 = new L2AirdropV2(
+            address(l2LiskToken), address(l2Claim), address(l2LockingPosition), ecosystemFundWalletAddress
+        );
+
+        bytes32[] memory merkleProof = new bytes32[](1);
+        vm.expectRevert("L2AirdropV2: airdrop has not started yet");
+        l2AirdropV2.claimAirdrop(bytes20(alice), 20 * 10 ** 18, merkleProof);
+    }
+
+    function test_ClaimAirdrop_AirdropOver() public {
+        // proceed time to HODLER_AIRDROPV2_DURATION + 1 so that airdrop period is over
+        vm.warp(block.timestamp + l2AirdropV2.HODLER_AIRDROPV2_DURATION() * 1 days + 1);
+
+        bytes32[] memory merkleProof = new bytes32[](1);
+        vm.expectRevert("L2AirdropV2: airdrop period is over");
+        l2AirdropV2.claimAirdrop(bytes20(alice), 20 * 10 ** 18, merkleProof);
+    }
+
+    function test_ClaimAirdrop_AmountIsZero() public {
+        bytes32[] memory merkleProof = new bytes32[](1);
+        vm.expectRevert("L2AirdropV2: amount is zero");
+        l2AirdropV2.claimAirdrop(bytes20(alice), 0, merkleProof);
+    }
+
+    function test_ClaimAirdrop_ZeroProofLength() public {
+        bytes32[] memory merkleProof = new bytes32[](0);
+        vm.expectRevert("L2AirdropV2: Merkle proof is empty");
+        l2AirdropV2.claimAirdrop(bytes20(alice), 20 * 10 ** 18, merkleProof);
+    }
+
+    function test_ClaimAirdrop_ZeroRecipientAddress() public {
```

Comment:
this test should be removed, because the eligibility should only be checked via the Merkle tree, not the claims contract.

Reply:
We need to have this test to check the recipient address to be not zero.